### ISSUE TITLE
Is there any possibility of presence of fully connected layer in YOLO V5

### ISSUE THREAD

**Issue by akshat-suwalka** [ultralytics/yolov5#2490]
## ❔Question
I am very new with the architecture of YOLO v5 and have a question regarding Fully connected layer.

Can we have by any means fully connected layer in a YOLO V5 or Are there any Fully connected layers present inside it?

Can we add a fully connected layer in the architecture by any means or technique without much affecting the model performance and Is it possible ?

If anybody have any idea or comments please let me know!

**Comment by cszer** [comment]
Hi , check yolov3 paper to understand perediction head more. 
Brief review - yolo doesnt predicit exclusive label for certain pixel in feature map, instead yolo allows as to have multiply classes. For some datasets that consist of few classes  and objects of different classes dont overlap much we can replace 1 dimension of fearture map to n dimensions (n means number of classes) and use softmax over it. Because yolo is one stage detector we dont have rpn and roi pooling stage, thats why fc layer is not needed to predict objects

**Comment by cszer** [comment]
Please explain more why you need fc layer in yolo

**Comment by akshat-suwalka** [comment]
@cszer Thanks for replying!

I have some architecture which can take input from fully connected layer only and its output is also the fully connected layer type.
I want to see that how can I add my architecture in yolov5 such that it satisfies the input-output condition of my architecture.

Do you have any idea?

**Comment by glenn-jocher** [comment]
@akshat-suwalka there are no nn.Linear() (or 'fully connected') layers in YOLOv5
https://pytorch.org/docs/stable/generated/torch.nn.Linear.html
